samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -179,20 +189,51 @@ public ApplicationStatus status(StreamApplication streamApp) {
   }
 
   /**
+   * Generates processorId for each {@link StreamProcessor} using the configured {@link ProcessorIdGenerator}
+   *
+   * @param config Application config
+   * @return String that uniquely represents an instance of {@link StreamProcessor} in the current JVM
+   */
+  /* package private */
+  String getProcessorId(Config config) {
```

Comment:
> let JC be the source of truth for these IDs.

Just to be clear, the JC within each StreamProcessor will be responsible for creating this processorId and registering. This is the current behavior and I hope everyone is in agreement with that. 
In case of yarn, the leader (aka AM) will assign processorIds through the environment. When we switch to use StreamProcessor with YarnJobCoordinator, the processorId generator will simply parse the processorId from the environment variable set by the AM.